lib/trycourier/brands.rb

```diff
@@ -0,0 +1,70 @@
+module Courier
+  class Brands
+    @@key = "/brands"
+
+    def initialize(session)
+      @session = session
+    end
+
+    def list(cursor: nil)
+      params = {}
+      if cursor
+        params["cursor"] = cursor
+      end
+      res = @session.send(@@key, "GET", params: params)
+      ErrorHandler.check_err(res)
+    end
+
+    def get(brand_id:)
+      path = @@key + "/" + brand_id.to_s
+      res = @session.send(path, "GET")
+      ErrorHandler.check_err(res)
+    end
+
+    def add(name:, settings:, id: nil, snippets: nil, idempotency_key: nil)
```

Comment:
Doesn't this syntax make name and settings required fields?

Reply:
It does -- which I would consider a feature and not a bug. There is a high amount of required KWargs throughout this that strengthen the contracts between all the method signatures, where if they didn't default to nil they would lead to you chasing down nil. The benefit is that you get an explicit, clear error over a nil bubbling through. If you really dont want them to be required, I would remove the presence of kwargs as a whole here because defaulting them to nil functionally gets you to the same point of having to chase it down later over see the clear exception.